### PR TITLE
core/cgroup: assigning empty string to Delegate= resets list of controllers

### LINKED ISSUE #7334: Delegate= has no easy way to set a smaller list
I'm not sure if this is really a problem…, I'm throwing this out here for discussion.

If we have a unit with `Delegate=yes`, to override this, it is necessary to set `Delegate=no` first, and then `Delegate=a b c`. We *could* make it so that if Delegate=yes would be followed by a different Delegate= setting, we'd reset the list before the new assignment. Or we could just document current semantics.

### PR DESCRIPTION
**Pull request by yuwata**
Before this, assigning empty string to `Delegate=` makes no change to the controller list. This is inconsistent to the other options that take list of strings. After this, when empty string is assigned to `Delegate=`, the list of controllers is reset. Such behavior is consistent to other options and useful for drop-in configs.

Closes #7334.

### CODE REVIEW

**Comment by yuwata** [comment]
@poettering I want to confirm that `delegate_controllers=0` is equivalent to `delegate=0`, is not it? Because when this PR is accepted, then there are no way to set `delegate=1` and `delegate_controllers=0`...

**Comment by poettering** [comment]
Hmm, I am puzzled by this? There should be a way to turn on delegation, but not delegate any controllers. How would you do that with your patch?

**Comment by yuwata** [comment]
@poettering Ah, I misunderstanded the code. The combination of delegate=1 and controllers=0 is meaningful...

**Comment by yuwata** [comment]
Maybe an appropriate way is that empty string resets delegate_controllers and enables delegation, as already documented. I will update this.

**Comment by yuwata** [comment]
@poettering Updated. PTAL.

File: src/core/dbus-cgroup.c [hunk 1]
```diff
@@ -360,33 +360,24 @@ static int bus_cgroup_set_transient_property(
                 return 1;
 
         } else if (streq(name, "DelegateControllers")) {
+                _cleanup_strv_free_ char **l = NULL;
+                char **p;
                 CGroupMask mask = 0;
 
-                r = sd_bus_message_enter_container(message, 'a', "s");
+                r = sd_bus_message_read_strv(message, &l);
```
Comment (poettering) [inline src/core/dbus-cgroup.c]: why this change? you don't actually need the list as char**, do you?

I mean, if all you want to check is whether an empty list is assigned, then mask == 0 is a good check, as we add a bit for every item we decode...

Comment (yuwata) [reply src/core/dbus-cgroup.c]: Indeed. Updated.

**Comment by yuwata** [comment]
I've updated this. PTAL.
